word_counter.rb

```diff
@@ -0,0 +1,18 @@
+class WordCounter
+  def self.count file  
+    raise 'WordCounter: no file given' unless file 
+    # use inject to create a hash of word counts
+    to_word_array(file).inject({}) { |h,w| h[w] ? h[w] += 1 : h[w] = 1; h }
```

Comment:
A couple tips for making this more readable:
 - Hashs can be instantiated with a default value. (`0` would be good choice here.)
 - If you switch from `inject` to `each_with_object`, you don't need to return the hash at the end of the block.